crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -980,6 +982,60 @@ export.get_balance
     # => [balance]
 end
 
+#! Returns the balance of a fungible asset associated with a faucet_id at the beginning of the 
+#! transaction.
+#!
+#! To get the initial balance this procedure applies the changes stored in the fungible asset vault
+#! delta to the current asset balance.
```

Comment:
Why do we need to use the vault delta rather than just the initial vault? It seems to me that it may be simpler just to read the value from the original vault - and I think we should the original have the initial vault root somewhere in the kernel memory.

Reply:
Makes sense, reworked it to use initial native account vault.